File: go/scrabble-score/scrabble_score.go
// Package scrabble provides a method to compute Scrable score.
package scrabble

import "strings"

func score(ch rune) int {
	switch ch {
	case 'D', 'G':
		return 2
	case 'B', 'C', 'M', 'P':
		return 3
	case 'F', 'H', 'V', 'W', 'Y':
		return 4
	case 'K':
		return 5
	case 'J', 'X':
		return 8
	case 'Q', 'Z':
		return 10
	default:
		return 1
	}
}

// Score computes the Scrabble score for given word.
func Score(word string) int {
	result := 0
	for _, ch := range strings.ToUpper(word) {
		result += score(ch)
	}
	return result
}

File: zig/scrabble-score/scrabble_score.zig
pub fn score(s: []const u8) u32 {
    var result: u32 = 0;

    for (s) |ch| {
        result += switch (toUpper(ch)) {
            'D', 'G' => 2,
            'B', 'C', 'M', 'P' => 3,
            'F', 'H', 'V', 'W', 'Y' => 4,
            'K' => 5,
            'J', 'X' => 8,
            'Q', 'Z' => 10,
            else => 1,
        };
    }

    return result;
}

fn toUpper(ch: u8) u8 {
    return switch (ch) {
        'a'...'z' => ch - 32,
        else => ch,
    };
}
